tests/test_input.py

```diff
@@ -208,6 +221,29 @@ def test_invert_xaxis(disc_xrotation):
     m, pos, vel, a = disc_xrotation
     gxchA = utils._get_rot_matrix(m, pos, vel)
 
-    # we want this to be the identity
-    invtest = a @ np.linalg.inv(gxchA)
-    np.testing.assert_allclose(invtest, np.identity(3), atol=1e-3, rtol=1e-4)

```

Comment:
Have you ditched this inverse methodology?

Reply:
Check first the `random` utilization, then check if this still fails.

Only if this is still failing, then move to testing component by component.

Reply:
Yes, because it worked for x-axis but didn't work for y-axis, I don't understand why this happens, but I think that it is because of the definition of the rotation matrix. This matrix makes two rotations, one on the z-axis and other on the x-axis, and by definition, the direction cosine of y-axis goes to zero leaving one degree of freedom. (I think that this should happen to the x-axis rotation too, but I don't find another explanation)

Reply:
Let's leave it like this for the moment